src/test/java/net/sf/jabref/importer/CustomImporterTest.java

```diff
@@ -0,0 +1,120 @@
+    }
+
+    @Test
+    public void testGetClassName() {
+        assertEquals("net.sf.jabref.importer.fileformat.CopacImporter", importer.getClassName());
+    }
+
+    @Test
+    public void testGetBasePath() {
+        assertEquals("src/main/java/net/sf/jabref/importer/fileformat/CopacImporter.java", importer.getBasePath());
+    }
+
+    @Test
+    public void testGetInstance() throws Exception {
+        assertEquals(new CopacImporter(), importer.getInstance());
+    }
+
+    @Test
+    public void testGetFileFromBasePath() {
+        assertEquals(new File("src/main/java/net/sf/jabref/importer/fileformat/CopacImporter.java"), importer.getFileFromBasePath());
+    }
+
+    @Test
+    public void testGetBasePathUrl() throws Exception {
+        assertEquals(new File("src/main/java/net/sf/jabref/importer/fileformat/CopacImporter.java").toURI().toURL(), importer.getBasePathUrl());
+    }
+
+    @Test
+    public void testGetAsStringList() {
+        assertEquals("Copac", importer.getAsStringList().get(0));
+        assertEquals("cpc", importer.getAsStringList().get(1));
+        assertEquals("net.sf.jabref.importer.fileformat.CopacImporter", importer.getAsStringList().get(2));
+        assertEquals("src/main/java/net/sf/jabref/importer/fileformat/CopacImporter.java", importer.getAsStringList().get(3));
+    }
+
+    @Test
+    public void testEqualsTrue() {
+        boolean test = importer.equals(importer);
+        assertTrue(test);
+    }
+
+    @Test
+    public void testEqualsFalse() {
+        boolean test = importer.equals(new CopacImporter());
+        assertFalse(test);
+    }
+
+    @Test
+    public void testCompareToSmaller() {
+        CustomImporter ovidImporter = new CustomImporter(new OvidImporter());
+
+        assertTrue(importer.compareTo(ovidImporter) < 0);
+    }
+
+    @Test
+    public void testCompareToGreater() {
+        CustomImporter bibtexmlImporter = new CustomImporter(new BibTeXMLImporter());
+        CustomImporter ovidImporter = new CustomImporter(new OvidImporter());
+
+        assertTrue(ovidImporter.compareTo(bibtexmlImporter) > 0);
+    }
+
+    @Test
+    public void testCompareToEven() {
+        assertEquals(0, importer.compareTo(importer));
+    }
+
+    @Test
+    public void testToString() {
+        assertEquals("Copac", importer.toString());
+    }
+
+    @Test
+    public void testConstructors(){
+        OvidImporter ovid = new OvidImporter();
+        List<String> dataTest = Arrays.asList(ovid.getFormatName(), ovid.getId(), ovid.getClass().getName(), "src/main/java/net/sf/jabref/importer/fileformat/OvidImporter.java");
+        assertEquals(new CustomImporter(dataTest), new CustomImporter(new OvidImporter()));
+
+        CopacImporter copac = new CopacImporter();
+        assertEquals(new CustomImporter(copac.getFormatName(), copac.getId(), copac.getClass().getName(), "src/main/java/net/sf/jabref/importer/fileformat/CopacImporter.java"), new CustomImporter(new CopacImporter()));
```

Comment:
Split in two test cases. TestListConstructor and testImporterConstructor or something like that. Also don't use copac.getFormatName(), use the value. Also for the others.